dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -129,182 +134,213 @@ public void run(final JobExecutionContext jobContext) throws JobExecutionExcepti
-            audit.setJobId(jobId);
-            audit.setJobName(dataMap.getString(QUARTZ_JOB_NAME));
-            audit.setLangList(UtilMethods.join(languageToIndex,","));
-            audit.setPath(paths.size() > 0 ? UtilMethods.join(paths,",") : "/*");
-            audit.setPathInclude(include);
-            siteSearchAuditAPI.save(audit);
-        }
-        catch(DotDataException ex) {
-            Logger.error(this, "can't save audit data",ex);
-        }
-        finally {
+                final SiteSearchAudit audit = new SiteSearchAudit();
+                audit.setPagesCount(pagesCount);
+                audit.setFilesCount(filesCount);
+                audit.setUrlmapsCount(urlmapCount);
+                audit.setAllHosts(preparedJobContext.isIndexAll());
+                audit.setFireDate(jobContext.getFireTime());
+                audit.setHostList(preparedJobContext.getJoinedHosts());
+                audit.setIncremental(preparedJobContext.isIncremental());
+                audit.setStartDate(preparedJobContext.getStartDate());
+                audit.setEndDate(preparedJobContext.getEndDate());
+                audit.setIndexName(
+                        UtilMethods.isSet(preparedJobContext.getNewIndexName()) ? preparedJobContext
+                                .getNewIndexName() : preparedJobContext.getIndexName());
+                audit.setJobId(preparedJobContext.getJobId());
+                audit.setJobName(preparedJobContext.getJobName());
+                audit.setLangList(preparedJobContext.getLangList());
+                audit.setPath(preparedJobContext.getPaths());
+                audit.setPathInclude(preparedJobContext.isPathInclude());
+                siteSearchAuditAPI.save(audit);
+
+
+        } catch (DotDataException ex) {
+            Logger.error(this, "can't save audit data", ex);
+        } finally {
             HibernateUtil.closeSession();
         }
-
         date = DateUtil.getCurrentDate();
         ActivityLogger.logInfo(getClass(), "Job Finished", "User: " +userAPI.getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME  );
         AdminLogger.log(getClass(), "Job Finished", "User: " +userAPI.getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME );
     }
 
+     private synchronized PreparedJobContext prepareJob(final JobExecutionContext jobContext)
+             throws DotDataException, IOException, DotSecurityException {
+
+         final JobDataMap dataMap = jobContext.getJobDetail().getJobDataMap();
+         String jobId = (String) dataMap.get(JOB_ID);
+         if (jobId == null) {
+             jobId = dataMap.getString(QUARTZ_JOB_NAME);
+         }
+
+         final boolean indexAll = UtilMethods.isSet((String) dataMap.get(INDEX_ALL));
+         final String[] indexHosts;
+         final Object obj = (dataMap.get(INDEX_HOST) != null) ? dataMap.get(INDEX_HOST) : new String[0];
```

Comment:
this could be handle in a method such as 

indexHosts = ConversionUtils.toArrayString (dataMap.get(INDEX_HOST));

Reply:
nah